sodym/stocks.py

```diff
@@ -34,6 +34,13 @@ def shape(self) -> tuple:
     def process_id(self) -> int:
         return self.process.id
 
+    def to_stock_type(self, desired_stock_type: type, **kwargs):
+        """Return an object of a new stock type with values and dimensions the same as the original.
+        **kwargs can be used to pass additional model attributes as required by the desired stock
+        type, if these are not contained in the original stock type.
+        """
+        return desired_stock_type(**self.__dict__, **kwargs)
```

Comment:
What does this __dict__ contain? Doesn't it also contain all the methods of the original type?

Reply:
good point, but these would any get ignored by pydantic when passed to into the new Stock

Reply:
I checked, and it doesn't contain the methods, at least not in classes derived from pydantic.BaseModel.

Reply:
It seems really unsafe and crude at first glance.
Stuff just gets lost along the way without a warning. 
But I suppose that's just the design in pydantic that one has to get used to, and it seems safe to trust they knew what they were doing...

Reply:
I actually like it, because if you pass a file with a bunch of stuff in, it just takes the things it wants and ignores the rest, while if it is missing something it wants, it complains.